tools/hangar/src/benchmarking/compare.ts

```diff
@@ -0,0 +1,135 @@
+import { readFileSync, appendFileSync } from "fs";
+import { parseRoundedJson } from "./util";
+import { getBenchForBranch, upsertPRComment } from "./github";
+import { createTable } from "./table_report";
+
+function avgBenches(benchList: any[]): Record<string, any> {
+  const mainBenchMeanOfMeans: Record<string, any> = {};
+  for (const bench of benchList) {
+    for (const item of bench) {
+      if (!mainBenchMeanOfMeans[item.name]) {
+        mainBenchMeanOfMeans[item.name] = [];
+      }
+      mainBenchMeanOfMeans[item.name].push(item["mean"]);
+    }
+  }
+  for (const key in mainBenchMeanOfMeans) {
+    const sum = mainBenchMeanOfMeans[key].reduce(
+      (a: number, b: number) => a + b,
+      0
+    );
+    mainBenchMeanOfMeans[key] = sum / mainBenchMeanOfMeans[key].length;
+  }
+
+  return mainBenchMeanOfMeans;
+}
+
+function getLocalBench(filename: string) {
+  try {
+    const benchData = parseRoundedJson(readFileSync(filename, "utf-8"));
+    return [benchData.testResults.compile];
+  } catch (e) {
+    console.warn(`Could not find local bench file ${filename}`);
+    return [];
+  }
+}
+
+async function getDataFromSource(source: string) {
+  if (source.startsWith(".") || source.startsWith("/")) {
+    return getLocalBench(source);
+  } else {
+    return await getBenchForBranch(source);
+  }
+}
+
+export async function compareBenchmarks(
+  previousSource: string,
+  targetSource: string
+) {
+  const previousResult = await getDataFromSource(previousSource);
+  const targetResult = await getDataFromSource(targetSource);
+
+  const previousBench = avgBenches(previousResult);
+  const targetBench = avgBenches(targetResult);
+
+  console.log(`${previousSource} => ${targetSource}`);
+
+  const differences: Record<string, any> = {};
+
+  for (const [itemName, newMean] of Object.entries(targetBench)) {
+    const oldMean = previousBench[itemName];
+
+    differences[itemName] = {};
+    differences[itemName].meanBefore = oldMean ?? NaN;
+    differences[itemName].meanAfter = newMean;
+    differences[itemName].meanDiff =
+      Math.round((newMean - oldMean) * 100) / 100;
+    differences[itemName].meanPercentDiff =
+      Math.round(((newMean - oldMean) / oldMean) * 10000) / 100;
+  }
+
+  const formatNumber = (num: number | typeof NaN | undefined, unit: string) => {
+    if (num === undefined || isNaN(num)) {
+      return "...";
+    }
+    return num.toFixed(2) + unit;
```

Comment:
It applies to all the measurements we finally print to the table: I think we can keep it at ms, no reason for 1/100 of ms precision.

Reply:
Ok I changed ms to be rounded, but I kept 1 decimal of precision of the % (to prevent it from saying 0% change)